synapse/lib/net.py

```diff
@@ -374,12 +374,14 @@ def rx(self, link, mesg):
                 return self.rxfunc(self, mesg)
             except Exception as e:
                 logger.exception('%s.rxfunc() failed on: %r' % (self.__class__.__name__, mesg))
+                self.fini()
```

Comment:
based on discussion, do we want to make these txfini() ?

Reply:
done